src/Microsoft.OData.Core/UriParser/SemanticAst/InNode.cs

```diff
@@ -50,11 +50,14 @@ public InNode(SingleValueNode left, CollectionNode right)
             this.left = left;
             this.right = right;
 
-            if (!this.left.GetEdmTypeReference().IsAssignableFrom(this.right.ItemType) &&
-                !this.right.ItemType.IsAssignableFrom(this.left.GetEdmTypeReference()))
+            if (!right.GetEdmTypeReference().IsUntyped())

```

Comment:
What should happen if right node is untyped?

Reply:
if right is untyped, then we don't have to validate the assignability, which is all the enclosed logic does.